res/shaders/rgbsignal.frag

```diff
@@ -51,7 +45,21 @@ void main(void) {
     // to show other things (e.g. the axes lines) even when we are on a pixel
     // that does not have valid waveform data.
     if (new_currentIndex >= 0 && new_currentIndex <= waveformLength - 1) {
-      vec4 new_currentDataUnscaled = getWaveformData(new_currentIndex) * allGain;
+        vec4 new_currentDataUnscaled;
+        if (splitStereoSignal) {
+            // Texture coordinates put (0,0) at the bottom left, so show the right
+            // channel if we are in the bottom half.
+            new_currentDataUnscaled = getWaveformData(uv.y < 0.5 ? new_currentIndex + 1 : new_currentIndex) * allGain;
+        } else {
+            vec4 leftDataUnscaled = getWaveformData(new_currentIndex);
+            vec4 rightDataUnscaled = getWaveformData(new_currentIndex + 1);
+            new_currentDataUnscaled = vec4(
+                                              max(leftDataUnscaled.x, rightDataUnscaled.x),
+                                              max(leftDataUnscaled.y, rightDataUnscaled.y),
+                                              max(leftDataUnscaled.z, rightDataUnscaled.z),
+                                              max(leftDataUnscaled.w, rightDataUnscaled.w)) *
+                    allGain;
+        }
```

Comment:
```suggestion
            new_currentDataUnscaled = max(leftDataUnscaled, rightDataUnscaled) * allGain;
        }
```

Reply:
Would this apply a max on each value independently? For example, would this capture (A.x, B.y, B.z, A.w)? 
We use a vec to carry a tuple of 4 values, but these gets carried inside a 4D vector for compatibility, however my naive impression was that `max` on `vec4` would pick either a vector or the other based on their normalization value.